app/uk/gov/hmrc/agentservicesaccount/controllers/desiDetails/ApplySACodeChangesController.scala

```diff
@@ -37,49 +36,34 @@ class ApplySACodeChangesController @Inject()(
                                               actions: Actions,
                                               val sessionCache: SessionCacheService,
                                               val acaConnector: AgentClientAuthorisationConnector,
-                                              applySaCodeChangesView: apply_sa_code_changes,
-                                              pcodRepository: PendingChangeOfDetailsRepository
+                                              applySaCodeChangesView: apply_sa_code_changes
                                             )(implicit appConfig: AppConfig,
                                               cc: MessagesControllerComponents,
-                                              ec: ExecutionContext) extends FrontendController(cc) with DesiDetailsJourneySupport with I18nSupport {
+                                              ec: ExecutionContext,
+                                              pcodRepository: PendingChangeOfDetailsRepository) extends FrontendController(cc) with DesiDetailsJourneySupport with I18nSupport {
```

Comment:
Why is this now an implicit parameter?

Reply:
Because it is used in trait DesiDetailsJourneySupport  by ifChangeContactFeatureEnabledAndNoPendingChanges(action: => Future[Result])
                                         (implicit request: AuthRequestWithAgentInfo[_],
                                          appConfig: AppConfig,
                                          pcodRepository: PendingChangeOfDetailsRepository,
                                          ec: ExecutionContext): Future[Result]

Reply:
I do not have to use implicit I can use val in trait

Reply:
That's fine, I was just curious